documentation/design/proposals/authn_authz_for_1.0.adoc

```diff
@@ -0,0 +1,329 @@
+# Proposal - Authentication and Authorization for 1.0
+
+## Overview
+
+1.0 will introduce a new Access Control Service.  This will replace the existing authentication services.
+It will have responsibility for identity management and authorization for the messaging use-case.
+
+For identity, it will have provision for:
+
+* leverage an external identity provider (such as a LDAP provider),
```

Comment:
External Keycloak?

Reply:
Yes, I think that would be useful for migrating from pre-1.0 versions as well.

Reply:
I've added Keycloak as a support external provider - GoCloak appears to have all the API we need to implement username/password auth (SCRAM-SHA SASL) and OAUTHBEARER in the same way we did in Java.